### ISSUE TITLE
Weapon Damage Not Updating

### ISSUE THREAD

**Issue by parinbhat23** [maxpaulus43/owlbear-shadowdark-character-sheet#45]
There's a small and annoying problem with the ShadowDark extension. All the weapon damages stay as a d6, no matter what I do to update them. I am using custom items for the weapons. In the item editor it shows the damages that I want, but when I hit "Update", it never updates it in the "Attacks" menu. It's super annoying. Can you fix it, please?

**Comment by maxpaulus43** [comment]
i'll take a look this week. thanks for posting an issue!

**Comment by maxpaulus43** [comment]
looks like I am able to make a d8 weapon:
![Screenshot 2024-03-17 at 11 08 21 PM](https://github.com/maxpaulus43/owlbear-shadowdark-character-sheet/assets/4512696/134cd3b6-6a9b-41bb-aeda-eb368f686b77)
![Screenshot 2024-03-17 at 11 08 04 PM](https://github.com/maxpaulus43/owlbear-shadowdark-character-sheet/assets/4512696/ee68cbe6-3aa6-472e-ad21-ee463d915b8d)

**Comment by maxpaulus43** [comment]
can you export your character and upload it to this issue?

**Comment by parinbhat23** [comment]
Here is my character. Please don't question the name or some of the stats - my players like to name themselves funky things. I have also included a screenshot of the damage it shows and the damage I set. This is a new issue - when I first had the extension, it worked fine.
<img width="397" alt="Weapon Damage" src="https://github.com/maxpaulus43/owlbear-shadowdark-character-sheet/assets/163665447/d1076ff4-0d83-4c6c-a05a-ea1cad2ca75f">
<img width="1470" alt="Weapon Fake Damage" src="https://github.com/maxpaulus43/owlbear-shadowdark-character-sheet/assets/163665447/d4faa2db-a94a-4f28-80be-dcd8cb8abf71">
[PUTINVsHITLERSheetU4.json](https://github.com/maxpaulus43/owlbear-shadowdark-character-sheet/files/14635647/PUTINVsHITLERSheetU4.json)

**Comment by maxpaulus43** [comment]
hmm i had a bit of time to look into this. I think the "Dragon's Demise" bonus is messing with the dicetype calculation. I will dig deeper tonight, but try removing that bonus, or turning it into a generic bonus

**Comment by maxpaulus43** [comment]
reason was because when determining which dice type to show in the attacks view, the tool was not considering if the weapon naturally had a better dice type than any diceType bonuses. fixed.

**Comment by maxpaulus43** [comment]
the dragon's demise bonus should be a generic bonus instead of a dicetype bonus. diceType bonuses will directly affect what diceType is shown in the attacks view.

**Comment by maxpaulus43** [comment]
side note: it's always interesting to see a character in "the wild". i notice that some custom items have 1000000000000000 for the value of "How Many of this item take up one slot?". Maybe I will update the UI to allow a checkbox that allows infinite items per slot.

**Comment by parinbhat23** [comment]
See, I only added so many custom items with crazy values because I think the Dnd encumbrance rule is stupid. I know it is realistic, but I don't want to factor gear weight into the game for my players. The extension uses "slots" to represent weight or storage space, and I don't want my players to have to leave stuff behind to carry other stuff. So, I made everything a custom item and made sure it can NEVER be placed in an item slot.
